YARA fails to build with OpenSSL 4.0

**Describe the bug**
YARA fails to build with OpenSSL 4.0 as the bundled authenticode-parser is incompatible.

**To Reproduce**
Steps to reproduce the behavior:
```
$ git clone https://github.com/VirusTotal/yara.git
$ cd yara
$ autoreconf -fiv
$ ./configure --with-crypto --enable-magic
$ make
....
libyara/modules/pe/authenticode-parser/certificate.c:68:47: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
   68 |         if (byte_array_init(&array, asn1String->data, asn1String->length) == -1)
      |                                               ^~
libyara/modules/pe/authenticode-parser/certificate.c:68:65: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
   68 |         if (byte_array_init(&array, asn1String->data, asn1String->length) == -1)
      |                                                                 ^~
libyara/modules/pe/authenticode-parser/certificate.c: In function 'certificate_new':
libyara/modules/pe/authenticode-parser/certificate.c:291:29: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  291 |     X509_NAME* issuerName = X509_get_issuer_name(x509);
      |                             ^~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/countersignature.c: In function 'pkcs9_countersig_new':
libyara/modules/pe/authenticode-parser/countersignature.c:144:32: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  144 |     ASN1_TYPE* messageDigest = PKCS7_get_signed_attribute(si, NID_pkcs9_messageDigest);
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/countersignature.c:150:57: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  150 |     size_t digestLen = messageDigest->value.octet_string->length;
      |                                                         ^~
libyara/modules/pe/authenticode-parser/countersignature.c:163:66: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  163 |     const uint8_t* digestData = messageDigest->value.octet_string->data;
      |                                                                  ^~
libyara/modules/pe/authenticode-parser/certificate.c:300:30: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  300 |     X509_NAME* subjectName = X509_get_subject_name(x509);
      |                              ^~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/countersignature.c:190:38: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  190 |     uint8_t* encData = si->enc_digest->data;
      |                                      ^~
libyara/modules/pe/authenticode-parser/countersignature.c:191:35: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  191 |     size_t encLen = si->enc_digest->length;
      |                                   ^~
libyara/modules/pe/authenticode-parser/countersignature.c:223:50: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  223 |             isValid = !memcmp(digest_info->digest->data, calc_digest, mdLen);
      |                                                  ^~
libyara/modules/pe/authenticode-parser/countersignature.c:238:36: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  238 |     calculate_digest(md, enc_digest->data, enc_digest->length, calc_digest);
      |                                    ^~
libyara/modules/pe/authenticode-parser/countersignature.c:238:54: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  238 |     calculate_digest(md, enc_digest->data, enc_digest->length, calc_digest);
      |                                                      ^~
libyara/modules/pe/authenticode-parser/countersignature.c: In function 'ms_countersig_impl_get_ts_tst_info_cms_':
libyara/modules/pe/authenticode-parser/countersignature.c:272:37: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  272 |     const uint8_t* data = (*content)->data;
      |                                     ^~
libyara/modules/pe/authenticode-parser/countersignature.c:273:71: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  273 |     TS_TST_INFO* ts_tst_info = d2i_TS_TST_INFO(NULL, &data, (*content)->length);
      |                                                                       ^~
libyara/modules/pe/authenticode-parser/countersignature.c: In function 'ms_countersig_impl_verify_digest_pkcs7_':
libyara/modules/pe/authenticode-parser/countersignature.c:362:5: warning: 'TS_VERIFY_CTX_set_store' is deprecated: Since OpenSSL 3.4;Unclear semantics, replace with TS_VERIFY_CTX_set0_store(). [-Wdeprecated-declarations]
  362 |     TS_VERIFY_CTX_set_store(ctx, store);
      |     ^~~~~~~~~~~~~~~~~~~~~~~
In file included from libyara/modules/pe/authenticode-parser/countersignature.c:32:
/usr/include/openssl/ts.h:434:13: note: declared here
  434 | X509_STORE *TS_VERIFY_CTX_set_store(TS_VERIFY_CTX *ctx, X509_STORE *s);
      |             ^~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/countersignature.c:364:5: warning: 'TS_VERIFY_CTX_set_certs' is deprecated: Since OpenSSL 3.4;Unclear semantics, replace with TS_VERIFY_CTX_set0_certs(). [-Wdeprecated-declarations]
  364 |     TS_VERIFY_CTX_set_certs(ctx, impl->funcs->get_certs(impl));
      |     ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/openssl/ts.h:442:17: note: declared here
  442 | STACK_OF(X509) *TS_VERIFY_CTX_set_certs(TS_VERIFY_CTX *ctx, STACK_OF(X509) *certs);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/countersignature.c:368:5: warning: 'TS_VERIFY_CTX_set_imprint' is deprecated: Since OpenSSL 3.4;Unclear semantics, replace with TS_VERIFY_CTX_set0_imprint(). [-Wdeprecated-declarations]
  368 |     TS_VERIFY_CTX_set_imprint(ctx, digest, digest_size);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/openssl/ts.h:427:16: note: declared here
  427 | unsigned char *TS_VERIFY_CTX_set_imprint(TS_VERIFY_CTX *ctx,
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/countersignature.c: In function 'ms_countersig_impl_verify_digest_cms_':
libyara/modules/pe/authenticode-parser/countersignature.c:403:26: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  403 |     if (ts_imprint_digest->length != (int)digest_size ||
      |                          ^~
libyara/modules/pe/authenticode-parser/countersignature.c:404:33: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  404 |         memcmp(ts_imprint_digest->data, digest, digest_size) != 0) {
      |                                 ^~
libyara/modules/pe/authenticode-parser/countersignature.c: In function 'ms_countersig_new':
libyara/modules/pe/authenticode-parser/countersignature.c:557:30: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  557 |     int digestLen = rawDigest->length;
      |                              ^~
libyara/modules/pe/authenticode-parser/countersignature.c:558:36: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  558 |     uint8_t* digestData = rawDigest->data;
      |                                    ^~
libyara/modules/pe/authenticode-parser/countersignature.c:574:36: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  574 |     calculate_digest(md, enc_digest->data, enc_digest->length, calc_digest);
      |                                    ^~
libyara/modules/pe/authenticode-parser/countersignature.c:574:54: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  574 |     calculate_digest(md, enc_digest->data, enc_digest->length, calc_digest);
      |                                                      ^~
make[1]: *** [Makefile:2070: libyara/modules/pe/authenticode-parser/la-certificate.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:2084: libyara/modules/pe/authenticode-parser/la-countersignature.lo] Error 1
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'get_content':
libyara/modules/pe/authenticode-parser/authenticode.c:81:47: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
   81 |     int len = content->d.other->value.sequence->length;
      |                                               ^~
libyara/modules/pe/authenticode-parser/authenticode.c:82:59: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
   82 |     const uint8_t* data = content->d.other->value.sequence->data;
      |                                                           ^~
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'parse_program_name':
libyara/modules/pe/authenticode-parser/authenticode.c:91:53: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
   91 |     const uint8_t* spcData = spcAttr->value.sequence->data;
      |                                                     ^~
libyara/modules/pe/authenticode-parser/authenticode.c:92:41: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
   92 |     int spcLen = spcAttr->value.sequence->length;
      |                                         ^~
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'parse_nested_authenticode':
libyara/modules/pe/authenticode-parser/authenticode.c:131:29: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  131 |         ASN1_TYPE* nested = X509_ATTRIBUTE_get0_type(attr, i);
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/authenticode.c:134:41: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  134 |         int len = nested->value.sequence->length;
      |                                         ^~
libyara/modules/pe/authenticode-parser/authenticode.c:135:53: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  135 |         const uint8_t* data = nested->value.sequence->data;
      |                                                     ^~
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'parse_pkcs9_countersig':
libyara/modules/pe/authenticode-parser/authenticode.c:162:29: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  162 |         ASN1_TYPE* nested = X509_ATTRIBUTE_get0_type(attr, i);
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/authenticode.c:165:41: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  165 |         int len = nested->value.sequence->length;
      |                                         ^~
libyara/modules/pe/authenticode-parser/authenticode.c:166:53: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  166 |         const uint8_t* data = nested->value.sequence->data;
      |                                                     ^~
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'parse_ms_countersig':
libyara/modules/pe/authenticode-parser/authenticode.c:193:29: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  193 |         ASN1_TYPE* nested = X509_ATTRIBUTE_get0_type(attr, i);
      |                             ^~~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/authenticode.c:196:41: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  196 |         int len = nested->value.sequence->length;
      |                                         ^~
libyara/modules/pe/authenticode-parser/authenticode.c:197:53: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  197 |         const uint8_t* data = nested->value.sequence->data;
      |                                                     ^~
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'authenticode_verify':
libyara/modules/pe/authenticode-parser/authenticode.c:212:79: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  212 |     const uint8_t* contentData = p7->d.sign->contents->d.other->value.sequence->data;
      |                                                                               ^~
libyara/modules/pe/authenticode-parser/authenticode.c:213:68: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  213 |     long contentLen = p7->d.sign->contents->d.other->value.sequence->length;
      |                                                                    ^~
libyara/modules/pe/authenticode-parser/authenticode.c: In function 'authenticode_new':
libyara/modules/pe/authenticode-parser/authenticode.c:318:42: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  318 |     int digestLen = messageDigest->digest->length;
      |                                          ^~
libyara/modules/pe/authenticode-parser/authenticode.c:319:54: error: invalid use of incomplete typedef 'ASN1_OCTET_STRING' {aka 'struct asn1_string_st'}
  319 |     const uint8_t* digestData = messageDigest->digest->data;
      |                                                      ^~
libyara/modules/pe/authenticode-parser/authenticode.c:366:25: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  366 |     ASN1_TYPE* digest = PKCS7_get_signed_attribute(si, NID_pkcs9_messageDigest);
      |                         ^~~~~~~~~~~~~~~~~~~~~~~~~~
libyara/modules/pe/authenticode-parser/authenticode.c:375:42: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  375 |     digestLen = digest->value.asn1_string->length;
      |                                          ^~
libyara/modules/pe/authenticode-parser/authenticode.c:376:43: error: invalid use of incomplete typedef 'ASN1_STRING' {aka 'struct asn1_string_st'}
  376 |     digestData = digest->value.asn1_string->data;
      |                                           ^~
libyara/modules/pe/authenticode-parser/authenticode.c:380:26: warning: initialization discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  380 |     ASN1_TYPE* spcInfo = PKCS7_get_signed_attribute(si, OBJ_txt2nid(NID_spc_info));
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~
make[1]: *** [Makefile:2063: libyara/modules/pe/authenticode-parser/la-authenticode.lo] Error 1
````

**Expected behavior**
YARA should build with OpenSSL 4.0.

**Screenshots**
Not Applicable.

**Please complete the following information:**
 - Arch Linux Rolling
 - YARA HEAD https://github.com/VirusTotal/yara/commit/129cdc43fbe29f477540e89c9f71086acb08913c

**Additional context**
 - https://github.com/avast/authenticode-parser/issues/23
 - https://sources.debian.org/data/main/y/yara/4.5.7-1/debian/patches/Use-OpenSSL-accessor-functions-instead-of-direct-field-ac.patch
 - [yara-openssl-4.0-build-failure.log](https://github.com/user-attachments/files/30387224/yara-openssl-4.0-build-failure.log)

@anthraxx @pierres

--- a/libyara/modules/pe/authenticode-parser/authenticode.c
+++ b/libyara/modules/pe/authenticode-parser/authenticode.c
@@ -78,8 +78,8 @@ static SpcIndirectDataContent* get_content(PKCS7* content)
     if (!spcContent)
         return NULL;
 
-    int len = content->d.other->value.sequence->length;
-    const uint8_t* data = content->d.other->value.sequence->data;
+    int len = ASN1_STRING_length(content->d.other->value.sequence);
+    const uint8_t* data = ASN1_STRING_get0_data(content->d.other->value.sequence);
 
     d2i_SpcIndirectDataContent(&spcContent, &data, len);
 
@@ -88,8 +88,8 @@ static SpcIndirectDataContent* get_content(PKCS7* content)
 
 static char* parse_program_name(ASN1_TYPE* spcAttr)
 {
-    const uint8_t* spcData = spcAttr->value.sequence->data;
-    int spcLen = spcAttr->value.sequence->length;
+    const uint8_t* spcData = ASN1_STRING_get0_data(spcAttr->value.sequence);
+    int spcLen = ASN1_STRING_length(spcAttr->value.sequence);
     SpcSpOpusInfo* spcInfo = d2i_SpcSpOpusInfo(NULL, &spcData, spcLen);
     if (!spcInfo)
         return NULL;
@@ -131,8 +131,8 @@ static void parse_nested_authenticode(PKCS7_SIGNER_INFO* si, AuthenticodeArray* 
         ASN1_TYPE* nested = X509_ATTRIBUTE_get0_type(attr, i);
         if (nested == NULL)
             break;
-        int len = nested->value.sequence->length;
-        const uint8_t* data = nested->value.sequence->data;
+        int len = ASN1_STRING_length(nested->value.sequence);
+        const uint8_t* data = ASN1_STRING_get0_data(nested->value.sequence);
         AuthenticodeArray* auth = authenticode_new(data, len);
         if (!auth)
             continue;
@@ -162,8 +162,8 @@ static void parse_pkcs9_countersig(PKCS7* p7, Authenticode* auth)
         ASN1_TYPE* nested = X509_ATTRIBUTE_get0_type(attr, i);
         if (nested == NULL)
             break;
-        int len = nested->value.sequence->length;
-        const uint8_t* data = nested->value.sequence->data;
+        int len = ASN1_STRING_length(nested->value.sequence);
+        const uint8_t* data = ASN1_STRING_get0_data(nested->value.sequence);
 
         Countersignature* sig = pkcs9_countersig_new(data, len, p7->d.sign->cert, si->enc_digest);
         if (!sig)
@@ -193,8 +193,8 @@ static void parse_ms_countersig(PKCS7* p7, Authenticode* auth)
         ASN1_TYPE* nested = X509_ATTRIBUTE_get0_type(attr, i);
         if (nested == NULL)
             break;
-        int len = nested->value.sequence->length;
-        const uint8_t* data = nested->value.sequence->data;
+        int len = ASN1_STRING_length(nested->value.sequence);
+        const uint8_t* data = ASN1_STRING_get0_data(nested->value.sequence);
 
         Countersignature* csig = ms_countersig_new(data, len, si->enc_digest);
         if (!csig)
@@ -209,8 +209,8 @@ static void parse_ms_countersig(PKCS7* p7, Authenticode* auth)
 
 static bool authenticode_verify(PKCS7* p7, PKCS7_SIGNER_INFO* si, X509* signCert)
 {
-    const uint8_t* contentData = p7->d.sign->contents->d.other->value.sequence->data;
-    long contentLen = p7->d.sign->contents->d.other->value.sequence->length;
+    const uint8_t* contentData = ASN1_STRING_get0_data(p7->d.sign->contents->d.other->value.sequence);
+    long contentLen = ASN1_STRING_length(p7->d.sign->contents->d.other->value.sequence);
 
     uint64_t version = 0;
     ASN1_INTEGER_get_uint64(&version, p7->d.sign->version);
@@ -315,8 +315,8 @@ AuthenticodeArray* authenticode_new(const uint8_t* data, int32_t len)
     int digestnid = OBJ_obj2nid(messageDigest->digestAlgorithm->algorithm);
     auth->digest_alg = strdup(OBJ_nid2ln(digestnid));
 
-    int digestLen = messageDigest->digest->length;
-    const uint8_t* digestData = messageDigest->digest->data;
+    int digestLen = ASN1_STRING_length(messageDigest->digest);
+    const uint8_t* digestData = ASN1_STRING_get0_data(messageDigest->digest);
     byte_array_init(&auth->digest, digestData, digestLen);
 
     SpcIndirectDataContent_free(dataContent);
@@ -372,8 +372,8 @@ AuthenticodeArray* authenticode_new(const uint8_t* data, int32_t len)
     digestnid = OBJ_obj2nid(si->digest_alg->algorithm);
     signer->digest_alg = strdup(OBJ_nid2ln(digestnid));
 
-    digestLen = digest->value.asn1_string->length;
-    digestData = digest->value.asn1_string->data;
+    digestLen = ASN1_STRING_length(digest->value.asn1_string);
+    digestData = ASN1_STRING_get0_data(digest->value.asn1_string);
     byte_array_init(&signer->digest, digestData, digestLen);
 
     /* Authenticode stores optional programName in non-optional SpcSpOpusInfo attribute */
@@ -551,7 +551,7 @@ AuthenticodeArray* parse_authenticode(const uint8_t* pe_data, uint64_t pe_len)
             continue;
         }
 
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
         int mdlen = EVP_MD_get_size(md);
 #else
         int mdlen = EVP_MD_size(md);

--- a/libyara/modules/pe/authenticode-parser/certificate.c
+++ b/libyara/modules/pe/authenticode-parser/certificate.c
@@ -31,7 +31,7 @@ SOFTWARE.
 
 #include "helper.h"
 
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
 /* Removes any escaping \/ -> / that is happening with oneline() functions
     from OpenSSL 3.0 */
 static void parse_oneline_string(char* string)
@@ -65,7 +65,7 @@ static void parse_name_attributes(X509_NAME* raw, Attributes* attr)
         const char* key = OBJ_nid2sn(OBJ_obj2nid(X509_NAME_ENTRY_get_object(entryName)));
 
         ByteArray array = {0};
-        if (byte_array_init(&array, asn1String->data, asn1String->length) == -1)
+        if (byte_array_init(&array, ASN1_STRING_get0_data(asn1String), ASN1_STRING_length(asn1String)) == -1)
             break;
 
         if (strcmp(key, "C") == 0 && !attr->country.data)
@@ -293,14 +293,14 @@ Certificate* certificate_new(X509* x509)
 
     result->issuer = strdup(buffer);
     /* This is a little ugly hack for 3.0 compatibility */
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
     parse_oneline_string(result->issuer);
 #endif
 
     X509_NAME* subjectName = X509_get_subject_name(x509);
     X509_NAME_oneline(subjectName, buffer, sizeof(buffer));
     result->subject = strdup(buffer);
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
     parse_oneline_string(result->subject);
 #endif
 
@@ -320,7 +320,7 @@ Certificate* certificate_new(X509* x509)
     EVP_PKEY* pkey = X509_get0_pubkey(x509);
     if (pkey) {
         result->key = pubkey_to_pem(pkey);
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
         result->key_alg = strdup(OBJ_nid2sn(EVP_PKEY_get_base_id(pkey)));
 #else
         result->key_alg = strdup(OBJ_nid2sn(EVP_PKEY_base_id(pkey)));

--- a/libyara/modules/pe/authenticode-parser/countersignature.c
+++ b/libyara/modules/pe/authenticode-parser/countersignature.c
@@ -147,7 +147,7 @@ Countersignature* pkcs9_countersig_new(
         goto end;
     }
 
-    size_t digestLen = messageDigest->value.octet_string->length;
+    size_t digestLen = ASN1_STRING_length(messageDigest->value.octet_string);
 
     if (!digestLen) {
         result->verify_flags = COUNTERSIGNATURE_VFY_DIGEST_MISSING;
@@ -160,7 +160,7 @@ Countersignature* pkcs9_countersig_new(
         goto end;
     }
 
-    const uint8_t* digestData = messageDigest->value.octet_string->data;
+    const uint8_t* digestData = ASN1_STRING_get0_data(messageDigest->value.octet_string);
     byte_array_init(&result->digest, digestData, digestLen);
 
     /* By this point we all necessary things for verification
@@ -187,8 +187,8 @@ Countersignature* pkcs9_countersig_new(
         goto end;
     }
 
-    uint8_t* encData = si->enc_digest->data;
-    size_t encLen = si->enc_digest->length;
+    uint8_t* encData = (uint8_t*) ASN1_STRING_get0_data(si->enc_digest);
+    size_t encLen = ASN1_STRING_length(si->enc_digest);
 
     /* Decrypt the encrypted digest */
     EVP_PKEY_verify_recover_init(ctx);
@@ -204,7 +204,7 @@ Countersignature* pkcs9_countersig_new(
     /* compare the encrypted digest and calculated digest */
     bool isValid = false;
 
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
     size_t mdLen = EVP_MD_get_size(md);
 #else
     size_t mdLen = EVP_MD_size(md);
@@ -220,7 +220,7 @@ Countersignature* pkcs9_countersig_new(
         const uint8_t* data_ptr = decData;
         DigestInfo* digest_info = d2i_DigestInfo(NULL, &data_ptr, decLen);
         if (digest_info) {
-            isValid = !memcmp(digest_info->digest->data, calc_digest, mdLen);
+            isValid = !memcmp(ASN1_STRING_get0_data(digest_info->digest), calc_digest, mdLen);
             DigestInfo_free(digest_info);
         } else {
             isValid = false;
@@ -235,7 +235,7 @@ Countersignature* pkcs9_countersig_new(
 
     /* Now check the countersignature message-digest that should correspond
      * to Signatures encrypted digest it countersigns */
-    calculate_digest(md, enc_digest->data, enc_digest->length, calc_digest);
+    calculate_digest(md, ASN1_STRING_get0_data(enc_digest), ASN1_STRING_length(enc_digest), calc_digest);
 
     /* Check if calculated one matches the stored one */
     if (digestLen != mdLen || memcmp(calc_digest, digestData, mdLen) != 0) {
@@ -269,8 +269,8 @@ TS_TST_INFO* IMPL_FUNC_NAME(get_ts_tst_info, cms)(CountersignatureImpl* impl)
         return NULL;
     }
 
-    const uint8_t* data = (*content)->data;
-    TS_TST_INFO* ts_tst_info = d2i_TS_TST_INFO(NULL, &data, (*content)->length);
+    const uint8_t* data = ASN1_STRING_get0_data(*content);
+    TS_TST_INFO* ts_tst_info = d2i_TS_TST_INFO(NULL, &data, ASN1_STRING_length(*content));
     if (!ts_tst_info) {
         return NULL;
     }
@@ -360,7 +360,7 @@ int IMPL_FUNC_NAME(verify_digest, pkcs7)(
 
     TS_VERIFY_CTX_set_flags(ctx, TS_VFY_VERSION | TS_VFY_IMPRINT);
     TS_VERIFY_CTX_set_store(ctx, store);
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
     TS_VERIFY_CTX_set_certs(ctx, impl->funcs->get_certs(impl));
 #else
     TS_VERIFY_CTS_set_certs(ctx, impl->funcs->get_certs(impl));
@@ -400,8 +400,8 @@ int IMPL_FUNC_NAME(verify_digest, cms)(
         return 0;
     }
 
-    if (ts_imprint_digest->length != (int)digest_size ||
-        memcmp(ts_imprint_digest->data, digest, digest_size) != 0) {
+    if (ASN1_STRING_length(ts_imprint_digest) != (int)digest_size ||
+        memcmp(ASN1_STRING_get0_data(ts_imprint_digest), digest, digest_size) != 0) {
         TS_TST_INFO_free(ts_tst_info);
         return 0;
     }
@@ -554,8 +554,8 @@ Countersignature* ms_countersig_new(const uint8_t* data, long size, ASN1_STRING*
 
     ASN1_STRING* rawDigest = TS_MSG_IMPRINT_get_msg(imprint);
 
-    int digestLen = rawDigest->length;
-    uint8_t* digestData = rawDigest->data;
+    int digestLen = ASN1_STRING_length(rawDigest);
+    uint8_t* digestData = (uint8_t*) ASN1_STRING_get0_data(rawDigest);
 
     byte_array_init(&result->digest, digestData, digestLen);
 
@@ -571,9 +571,9 @@ Countersignature* ms_countersig_new(const uint8_t* data, long size, ASN1_STRING*
     }
 
     uint8_t calc_digest[EVP_MAX_MD_SIZE];
-    calculate_digest(md, enc_digest->data, enc_digest->length, calc_digest);
+    calculate_digest(md, ASN1_STRING_get0_data(enc_digest), ASN1_STRING_length(enc_digest), calc_digest);
 
-#if OPENSSL_VERSION_NUMBER >= 0x3000000fL
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
     int mdLen = EVP_MD_get_size(md);
 #else
     int mdLen = EVP_MD_size(md);

